keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/Differentiable.java

```diff
@@ -0,0 +1,28 @@
+package io.improbable.keanu.vertices.dbl.probabilistic;
+
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableList;
+
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.diff.DualNumber;
+
+public interface Differentiable {
+
+    DualNumber calculateDualNumber(Map<Vertex, DualNumber> dualNumbers);
+
+    default DualNumber getDualNumber() {
+        return new Differentiator().calculateDual((Vertex & Differentiable) this);
```

Comment:
this `new` is likely to lead to memory allocation - and may be more efficient as a static method...

Reply:
I think the GC will happily take care of this kind of memory allocation, won't it? But you're right, in this case the object has no state so there's no reason not to make `calculateDual` static - I will change it.